server/src/main/java/io/spine/server/model/HandlerMap.java

```diff
@@ -220,14 +223,34 @@ public H handlerOf(M messageClass) {
 
     private H checkSingle(Collection<H> handlers, M targetType) {
         int count = handlers.size();
-        checkState(count == 1,
-                   "Unexpected number of handlers for messages of class %s: %s.%n%s",
-                   targetType, count, handlers);
-        H result = handlers
-                .stream()
-                .findFirst()
-                .get();
-        return result;
+        if (count != 1) {
+            throw wrongMethodCount(handlers, targetType);
+        }
+        H handler = Iterables.getOnlyElement(handlers);
+        return handler;
+    }
+
+    private IllegalStateException wrongMethodCount(Collection<H> handlers, M targetType) {
+        int count = handlers.size();
+        if (count == 0) {
+            _error().log("No handler method found for the type `%s`.", targetType);
+            throw newIllegalStateException(
+                    "Unexpected number of handlers for messages of class %s: %d.%n%s",
+                    targetType, count, handlers
+            );
+        } else {
+            /*
+              The map should have found all the duplicates during construction.
+              This is a fail-safe execution branch which ensures that no changes in the `HandlerMap`
+              implementation corrupt the model.
+            */
+            _error().log(
+                    "There are %d handler methods found for the type `%s`." +
+                            "Expected only one. Model is corrupt.",
```

Comment:
> Model is corrupt

This doesn't tell the reader much of info. It just scares him. Tell the reader what to do about the problem.
Please include method names into diagnostics.